File: laba2/task3.cs
using System;

public class Kata
{
    public static string WhatCentury(string year)
    {
        int y = int.Parse(year);
        int century = (y + 99) / 100;
        
        if (century % 100 >= 11 && century % 100 <= 13)
            return century + "th";
            
        switch (century % 10)
        {
            case 1: return century + "st";
            case 2: return century + "nd";
            case 3: return century + "rd";
            default: return century + "th";
        }
    }
}

File: laba2/task3.js
function whatCentury(year) {
  const n = Math.floor((+year - 1) / 100) + 1;
  const s = n.toString();
  if (/(11|12|13)$/.test(s)) return s + "th";
  if (s.endsWith("1")) return s + "st";
  if (s.endsWith("2")) return s + "nd";
  if (s.endsWith("3")) return s + "rd";
  return s + "th";
}
